src/triage/component/architect/feature_group_mixer.py

```diff
@@ -31,6 +32,23 @@ def leave_one_out(feature_groups):
         results.append(feature_dict)
     return results
 
+def all_combinations(feature_groups):
+    """Return all combinations of groups, excluding repeated groups
+
+    Args:
+        feature_groups (list) The feature groups to apply the strategy to
+
+    Returns: A list of feature dicts
+    """
+    results = []
+    for number_feature_groups in range(0, len(feature_groups) + 1):
+        for combo in itertools.combinations(feature_groups,\
+                                            number_feature_groups):
+            feature_dict = FeatureGroup()
+            for group_element in combo:
+                feature_dict.update(group_element)
+            results.append(feature_dict)
+    return list(filter(None, results))
```

Comment:
…And then this can just return:

```suggestion
    return results
```